DependencyInjection/CompilerPass/RegisterCompilerPass.php

```diff
@@ -107,31 +97,24 @@ private function getClasses($directory)
      *
      * @param ContainerBuilder $container
      * @param string           $className
+     * @param array            $tags
      */
-    private function registerClass(ContainerBuilder $container, $prefix, $className)
+    private function registerClass(ContainerBuilder $container, $className, $tags)
     {
-        $id = sprintf('%s.%s', $prefix, $className);
-
-        if ($container->has($id)) {
+        if ($container->has($className)) {
             return;
         }
 
-        $definition = $container->register($id, $className);
+        $definition = $container->register($className, $className);
         $definition->setAutowired(true);
 
-        $this->customizeServiceDefinition($container, $definition);
-    }
+        foreach ($tags as $tagClassName => $classTags) {
+            if ($className !== $tagClassName && !is_subclass_of($className, $tagClassName)) {
```

Comment:
I've been thinking about this and maybe ``is_a($className, $tagClassName, true`` would be simpler here, WDYT?